src/ESLint.Formatter/sarif.js

```diff
@@ -179,9 +181,12 @@ module.exports = function (results, data) {
                     }
 
                     if (message.line > 0 || message.column > 0) {
-                        sarifRepresentation.locations[0].physicalLocation.region = {
-                            startLine: message.line,
-                            startColumn: message.column
+                        sarifRepresentation.locations[0].physicalLocation.region = {};
+                        if (message.line > 0) {
+                            sarifRepresentation.locations[0].physicalLocation.region.startLine = message.line;
+                        }
```

Comment:
>} [](start = 24, length = 1)

This change is fine. Or, to be precise, it's an improvement over the existing code. Be aware that if a SARIF region is being represented with line/column properties (SARIF can represent regions in other ways, such as offset/length), then the `startLine` property is required. So if you did have a message with a valid column but without a valid line, then the resulting SARIF file would be invalid. Having said that -- again, you're no worse off than before.

Reply:
That makes sense - though it's probably less common that a tool outputs a column without a line. Thanks